src/interface_adapter/round/RoundState.java

```diff
@@ -10,6 +13,13 @@ public class RoundState {
     private int currentLives;
     private int maxRounds;
     private int currentRoundNumber;
+    private String multipleChoice1 = "";
+    private String multipleChoice2 = "";
+    private String multipleChoice3 = "";
+    private String multipleChoice4 = "";
+    private boolean isEasyRound = false;
+    private boolean isMediumRound = false;
+    private boolean isHardRound = false;
```

Comment:
```suggestion
```
Keeping separate flags for things like this is generally a bad idea for maintainability. In essence, we just need to know if there is a multiple-choice round or not, but this can be derived from the size of `multipleChoiceOptions`, i.e. if it is non-empty then it is a multiple choice round, otherwise it's not. See next comments for further clarification.